File: test/language/testdata/java/StepDefinitions.java
import io.cucumber.java.en.And;
import io.cucumber.java.en.But;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.When;
import io.cucumber.java.en.Then;

public class StepDefinitions {
    @Given("a {uuid}")
        void a_uuid(String uuid) {
    }

    @When("a {date}")
        void a_date(Date date) {
    }

    @Then("a {planet}")
        void a_date(Date date) {
    }

    @And("^a regexp$")
        void a_regexp() {
    }

    @But("an {undefined-parameter}")
        void an_undefined_parameter(Date date) {
    }

    @Given(
        "the " + "bee's "
        + "knees")
        void the_bees_knees(Date date) {
    }
}

File: test/language/testdata/rust/StepDefinitions.rs
use cucumber::{given, then, when, World};


#[derive(Debug, Default, World)]
struct StepDefinitions {}

#[given(expr = "a {uuid}")]
fn a_uuid(world: &mut StepDefinitions, uuid: String) {}

#[given(expr = "a {date}")]
fn a_date(world: &mut StepDefinitions,date: chrono::DateTime) {}

mod nested {
    #[given("a {planet}")]
    fn a_date(world: &mut StepDefinitions,date: chrono::DateTime) {}
}

#[cucumber::given(regex = r"^a regexp$")]
fn a_regexp(world: &mut StepDefinitions) {}

#[given(expr = "an {undefined-parameter}")]
fn an_undefined_parameter(world: &mut StepDefinitions, date: chrono::DateTime) {}

#[given("the bee's knees")]
/// A doc comment between the attribute and the fn
#[when("a stacked step")]
fn the_bees_knees(world: &mut StepDefinitions) {}
